### DISCUSSION TITLE
About Bitrot Protection query

### DISCUSSION THREAD

**Discussion by Hello-Linux** [minio/minio#19443] [Q&A]
Here is doc show

**MinIO’s optimized implementation of the [HighwayHash algorithm](https://github.com/minio/highwayhash/blob/master/README.md) ensures that it captures and heals corrupted objects on the fly. Integrity is ensured from end to end by computing a hash on READ and verifying it on WRITE from the application, across the network, and to the memory or drive. The implementation is designed for speed and can achieve hashing speeds over 10 GB/sec on a single core on Intel CPUs.**

We have a special use case here, minio uses the underlying disk driver is a file system mount point we developed such as /mnt/disk{0,12,3,}, I read minio official documentation says there is a Bitrot Protection mechanism, We would like to know is this function performed regularly scan? Or will we start scanning when our client initiates a read/write request? We want to know this function mainly because the directory of this mount point is our self-developed file system. It will monitor whether the client has initiated a read request to the file in the current directory, and if it has detected it, it will trigger some operations

**Comment by klauspost** [comment]
>  We would like to know is this function performed regularly scan?  Or will we start scanning when our client initiates a read/write request?

These questions make no sense.

If you are doing your own modification I suggest you start diving into the source code.

**Reply by Hello-Linux** [reply]
Thank you for your answer. I just want to wait until we get to this. Is this fix a timed scan or a dynamic capture and repair of damaged objects every time I read or write?

**Reply by klauspost** [reply]
Both. We have an async scanner that does heal checks and we fire off async healing operations if we see bad/missing data when reading.

**Reply by Hello-Linux** [reply]
Can this async scanner  be turned off? For example, we don't want minio to scan regularly

**Reply by klauspost** [reply]
No. It performs needed maintenance.

**Reply by Hello-Linux** [reply]
Ok, thanks! In addition, I would like to know how often this async scanner is executed on a regular basis? Once a day or。。。。。

**Reply by klauspost** [reply]
Continously. https://min.io/docs/minio/kubernetes/upstream/administration/object-management/object-lifecycle-management.html#lifecycle-management-object-scanner

**Reply by Hello-Linux** [reply]
ok,thank you,but  we want to know this function mainly because the directory of this mount point is our self-developed file system. It will monitor whether the client has initiated a read request to the file in the current directory, and if it has detected it, it will trigger some operations,We wondered if this bitrot protection would trigger the detection mechanism of our own file system

**Reply by klauspost** [reply]
@Hello-Linux I can't tell. You can just download the server and try it out.

**Reply by Hello-Linux** [reply]
Let's side the async scanner will call some file operations function, such as posix open, read, write, lseek, close function？Our underlying File System is called the linux file system DMAPI(Dynamic File System Management API) interface, I don't know if we are also called this, if the call is this may trigger our file system special operations.

**Reply by klauspost** [reply]
Your filesystem = your problem.

**Reply by Hello-Linux** [reply]
ok thank you
